// @fitbit/sdk/lib/buildTargets.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.generateBuildTargets = void 0;
const tslib_1 = require("tslib");
const lodash_1 = tslib_1.__importDefault(require("lodash"));
const semver_1 = tslib_1.__importDefault(require("semver"));
const sdkVersion_1 = tslib_1.__importDefault(require("./sdkVersion"));
const image_codec_txi_1 = require("@fitbit/image-codec-txi");
const baseBuildTargets = {
    rhea: {
        displayName: 'Fitbit Sense 2',
        minSDKVersion: '7.0.0',
        platform: ['128.1.1+'],
        resourceFilterTag: '336x336',
        defaultTXIOutputFormat: image_codec_txi_1.TXIOutputFormat.RGBA4444,
        specs: {
            screenSize: {
                width: 336,
                height: 336,
            },
        },
    },
    hera: {
        displayName: 'Fitbit Versa 4',
        minSDKVersion: '7.0.0',
        platform: ['128.1.1+'],
        resourceFilterTag: '336x336',
        defaultTXIOutputFormat: image_codec_txi_1.TXIOutputFormat.RGBA4444,
        specs: {
            screenSize: {
                width: 336,
                height: 336,
            },
        },
    },
};
let extraBuildTargets;
try {
    extraBuildTargets = require('@fitbit/sdk-build-targets').default;
}
catch (_a) { }
function generateBuildTargets() {
    return lodash_1.default.pickBy(Object.assign(Object.assign({}, baseBuildTargets), extraBuildTargets), ({ minSDKVersion }) => minSDKVersion === undefined ||
        semver_1.default.gte((0, sdkVersion_1.default)().format(), minSDKVersion));
}
exports.generateBuildTargets = generateBuildTargets;
const buildTargets = generateBuildTargets();
exports.default = buildTargets;
